libraries/AP_HAL_ChibiOS/UARTDriver.cpp

```diff
@@ -1234,6 +1238,68 @@ void UARTDriver::_rx_timer_tick(void)
     if (_wait.thread_ctx && _readbuf.available() >= _wait.n) {
         chEvtSignal(_wait.thread_ctx, EVT_DATA);
     }
+#if HAL_FORWARD_OTG2_SERIAL
+    // this code is used to forward data from a serial port to the USB
+    // Used for connecting to Secondary Autopilot to communicate over
+    // USB, for firmware updates, configuration etc
+    if (sdef.get_index() == HAL_FORWARD_OTG2_SERIAL) {
+        if (lock_read_key == HAL_FORWARD_OTG2_SERIAL_LOCK_KEY &&
+            lock_write_key == HAL_FORWARD_OTG2_SERIAL_LOCK_KEY &&
+            SDU2.config->usbp->state == USB_ACTIVE &&
+            hal.analogin->power_status_flags() | MAV_POWER_STATUS_USB_CONNECTED) {
```

Comment:
needs to be an & not an |

Reply:
+ consider to add [PVS-Studio](https://pvs-studio.com/en/pvs-studio/) static code analyser to spot such type of issues. In their publications as a part of marketing activities it was shown on open source code repos it is good with such issues.